lib/src/common/model/checks.dart

```diff
@@ -0,0 +1,407 @@
+import 'package:github/src/util.dart';
+import 'package:meta/meta.dart';
+
+@immutable
+abstract class _EnumWithValue {
```

Comment:
This seems like a fairly general utility that could be reused. Would you be opposed to moving it to utils.dart and making a public class?